### PR TITLE
packet: fix memory leak in `packet_x11_open()`

### PR DESCRIPTION
**Pull request by vszakats**
Reported-by: 尹麓鸣
Ref: https://github.com/LuMingYinDetect/libssh_defects
Closes #xxxx

---

/cc @LuMingYinDetect

### CODE REVIEW

File: src/packet.c [hunk 1]
```diff
@@ -405,6 +405,7 @@ packet_x11_open(LIBSSH2_SESSION * session, unsigned char *data,
             rc = _libssh2_transport_send(session, x11open_state->packet, 17,
                                          NULL, 0);
             if(rc == LIBSSH2_ERROR_EAGAIN) {
+                LIBSSH2_FREE(session, channel);
```
Comment (willco007) [inline src/packet.c:408]: If we free `channel` here, what happens when it comes back in to this this function after EAGAIN is hit and the state is currently set to `libssh2_NB_state_created` which assumes channel is already allocated and usable?

Comment (willco007) [reply src/packet.c:408]: Secondly, just freeing channel wouldn't be enough, you'd also need to free `channel->channel_type`. That said, I don't think this is a leak.

Comment (vszakats) [reply src/packet.c:408]: Thanks @willco007!

**Comment by vszakats** [comment]
Closing without merge.

**Pull request closed without merging**